workers/unity/Packages/io.improbable.gdk.gameobjectcreation/Tests/Editmode/GameObjectCreationTests.cs

```diff
@@ -0,0 +1,153 @@
+        {
+            World.Dispose();
+        }
+
+        [Test]
+        public void Create_gameobject_after_required_components_arrive_in_same_frame()
+        {
+            World
+                .Step(world =>
+                {
+                    world.Connection.CreateEntity(entityId, new EntityTemplate());
+
+                    world.Connection.AddComponent(entityId, Position.ComponentId,
+                        new Position.Update { Coords = Coordinates.Zero });
+
+                    world.Connection.AddComponent(entityId, Metadata.ComponentId,
+                        new Metadata.Update { EntityType = "TestObject" });
+
+                    var gameObjectExists = linker.EntityIdToGameObjects
+                        .TryGetValue(new EntityId(entityId), out var gameObjects);
+
+                    Assert.IsFalse(gameObjectExists);
+                })
+                .Step(world =>
+                {
+                    var gameObjectExists = linker.EntityIdToGameObjects
+                        .TryGetValue(new EntityId(entityId), out var gameObjects);
+
+                    Assert.IsTrue(gameObjectExists);
+                });
+        }
+
+        [Test]
+        public void Create_gameobject_after_required_components_arrive_in_multiple_frames()
+        {
+            World
+                .Step(world =>
+                {
+                    world.Connection.CreateEntity(entityId, new EntityTemplate());
+                })
+                .Step(world =>
+                {
+                    world.Connection.AddComponent(entityId, Position.ComponentId,
+                        new Position.Update { Coords = Coordinates.Zero });
+
+                    var gameObjectExists = linker.EntityIdToGameObjects
+                        .TryGetValue(new EntityId(entityId), out var gameObjects);
+
+                    Assert.IsFalse(gameObjectExists);
+                })
+                .Step(world =>
+                {
+                    world.Connection.AddComponent(entityId, Metadata.ComponentId,
+                        new Metadata.Update { EntityType = "TestObject" });
+
+                    var gameObjectExists = linker.EntityIdToGameObjects
+                        .TryGetValue(new EntityId(entityId), out var gameObjects);
+
+                    Assert.IsFalse(gameObjectExists);
+                })
+                .Step(world =>
+                {
+                    var gameObjectExists = linker.EntityIdToGameObjects
+                        .TryGetValue(new EntityId(entityId), out var gameObjects);
+
+                    Assert.IsTrue(gameObjectExists);
+                });
+        }
+
+        private class TestGameObjectCreator : IEntityGameObjectCreator
+        {
+            private readonly string workerType;
+
+            private readonly Dictionary<EntityId, GameObject> entityIdToGameObject = new Dictionary<EntityId, GameObject>();
+
+            public ComponentType[] MinimumComponentTypes => new[]
```

Comment:
I think this could be replaced with something like
`public ComponentType[] MinimumComponentTypes { get; } = new[]`
To make it a readonly variable.
It wont matter much since we only read from this once, but up for consideration

Reply:
included it